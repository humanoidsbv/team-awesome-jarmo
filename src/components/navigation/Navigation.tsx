import Link from "next/link";
import * as Styled from "./Navigation.styled";

type MenuProps = {
  isActive: boolean;
};

export const Navigation = ({ isActive }: MenuProps) => {
  const menuItems = [
    { name: "Timesheet", link: "/", id: 1 },
    { name: "Team Members", link: "/team-members", id: 2 },
    { name: "Projects", link: "/projects", id: 3 },
    { name: "Clients", link: "/client", id: 4 },
    { name: "Documents", link: "/documents", id: 5 },
  ];

  return (
    <Styled.Nav isActive={isActive}>
      <Styled.MenuList data-cy="menu">
        {menuItems.map((menuItem) => (
          <Styled.MenuItem key={menuItem.id} data-cy="menu-item">
            <Link href={menuItem.link} passHref>
              <Styled.Link>{menuItem.name}</Styled.Link>
            </Link>
          </Styled.MenuItem>
        ))}
      </Styled.MenuList>
    </Styled.Nav>
  );
};
