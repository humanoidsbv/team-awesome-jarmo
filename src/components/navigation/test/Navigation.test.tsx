import React from "react";
import renderer from "react-test-renderer";
import { Navigation } from "../Navigation";

it("renders correctly", () => {
  const tree = renderer.create(<Navigation isActive />).toJSON();
  expect(tree).toMatchSnapshot();
});
