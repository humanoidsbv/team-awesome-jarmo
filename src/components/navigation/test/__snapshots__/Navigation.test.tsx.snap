// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Navigationstyled__Nav-sc-e77p0y-0 fuFDAf"
>
  <ul
    className="Navigationstyled__MenuList-sc-e77p0y-1 jVPQRI"
  >
    <li
      className="Navigationstyled__MenuItem-sc-e77p0y-3 khhQbA"
    >
      <a
        className="Navigationstyled__Link-sc-e77p0y-2 RDTAP"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Timesheet
      </a>
    </li>
    <li
      className="Navigationstyled__MenuItem-sc-e77p0y-3 khhQbA"
    >
      <a
        className="Navigationstyled__Link-sc-e77p0y-2 RDTAP"
        href="/team-members"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Team Members
      </a>
    </li>
    <li
      className="Navigationstyled__MenuItem-sc-e77p0y-3 khhQbA"
    >
      <a
        className="Navigationstyled__Link-sc-e77p0y-2 RDTAP"
        href="/projects"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Projects
      </a>
    </li>
    <li
      className="Navigationstyled__MenuItem-sc-e77p0y-3 khhQbA"
    >
      <a
        className="Navigationstyled__Link-sc-e77p0y-2 RDTAP"
        href="/client"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Clients
      </a>
    </li>
    <li
      className="Navigationstyled__MenuItem-sc-e77p0y-3 khhQbA"
    >
      <a
        className="Navigationstyled__Link-sc-e77p0y-2 RDTAP"
        href="/documents"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Documents
      </a>
    </li>
  </ul>
</div>
`;
